UM/PluginRegistry.py

```diff
@@ -70,6 +71,21 @@ def __init__(self, application: "Application", parent: QObject = None) -> None:
 
         self._supported_file_types = {"umplugin": "Uranium Plugin"} # type: Dict[str, str]
 
+        self._check_if_trusted = False  # type: bool
+        self._checked_plugin_ids = []     # type: List[str]
+        self._distrusted_plugin_ids = []  # type: List[str]
+        self._trust_checker = None  # type: Optional[Trust]
+
+    def setCheckIfTrusted(self, check_if_trusted: bool) -> None:
+        self._check_if_trusted = check_if_trusted
+        if self._check_if_trusted:
+            self._trust_checker = Trust.getInstance()
+            if not self._trust_checker:
+                raise Exception("Aborting! Couldn't find root public key '{0}'!".format(Trust.getPublicRootKeyPath()))
```

Comment:
I see your point: Trust might fail for other reasons in the future. So, I've changed it: There are now two ways to get an instance of Trust: `getInstance` and `getInstanceOrNone`. I've removed the raised exception here (and documented that it can crash and that the crash is intended if it happens).

This is done in UM 27562aa3.